File: source-connector/src/test/java/sourceconnector/domain/log/JSONLogTest.java
package sourceconnector.domain.log;

import offsetmanager.domain.file.LocalFileKey;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import java.nio.file.Path;

import static org.assertj.core.api.Assertions.assertThat;

class JSONLogTest {

  @DisplayName("JSONLog has a payload and fileMetadata")
  @Test
  void jsonLogTest() {
    // given
    JSONLog jsonLog = new JSONLog(
      "log payload",
      new FileLogMetadata(LocalFileKey.from(Path.of("file.ndjson")), 0L)
    );
    // when then
    assertThat(jsonLog.get()).isEqualTo("log payload");

    LogMetadata metadata = jsonLog.getMetadata();
    assertThat(metadata.key()).isEqualTo(LocalFileKey.from(Path.of("file.ndjson")));
    assertThat(metadata.offset()).isEqualTo(0L);
  }

}

File: source-connector/src/test/kotlin/sourceconnector/domain/log/JSONLogTest.kt
package sourceconnector.domain.log

import offsetmanager.domain.file.FileKey
import offsetmanager.domain.file.LocalFileKey.Companion.from
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test
import java.nio.file.Path

internal class JSONLogTest {
  @DisplayName("JSONLog has a payload and fileMetadata")
  @Test
  fun jsonLogTest() {
    // given
    val jsonLog = JSONLog(
      "log payload",
      FileLogMetadata(from(Path.of("file.ndjson")), 0L)
    )
    // when then
    Assertions.assertThat(jsonLog.get()).isEqualTo("log payload")

    val metadata = jsonLog.metadata
    Assertions.assertThat<FileKey>(metadata.key).isEqualTo(from(Path.of("file.ndjson")))
    Assertions.assertThat(metadata.offset).isEqualTo(0L)
  }
}
